### PR TITLE
Closes #18068: Use AwesomeBarFacts for AwesomeBar Telemetry

### LINKED ISSUE #18068: Use AwesomeBarFacts for AwesomeBar Telemetry
Collect Awesome Bar facts to be included in the Glean report.

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by rocketsroger** [comment]
### **Request for data collection review**
1. What questions will you answer with this data?
More understanding of user behaviour so we can refine suggestions that are shown in the Awesomebar. 
2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements? 
This allows for a better Awesomebar suggestions.
3. What alternative methods did you consider to answer these questions? Why were they not sufficient? 
No alternative is available to answer this question.
4. Can current instrumentation answer these questions? 
No current instrumentation can answer this question.
5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki.

Measurement Description | Data Collection Category | Tracking Bug #
-- | -- | --
Awesomebar bookmark suggestion clicked | category 2 -   interaction data | #18068
Awesomebar clipboard suggestion clicked | category 2 -   interaction data | #18068
Awesomebar history suggestion clicked | category 2 -   interaction data | #18068
Awesomebar search action clicked | category 2 -   interaction data | #18068
Awesomebar search suggestion clicked | category 2 -   interaction data | #18068
Awesomebar opened tab suggestion clicked | category 2 -   interaction data | #18068

6. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way. 
https://github.com/mozilla-mobile/fenix/blob/master/docs/metrics.md
7. How long will this data be collected? Choose one of the following: 
This is scoped to a time-limited experiment/project until date 08-01-2021.
8. What populations will you measure? 
No filters - all locales, channels, etc.
9. If this data collection is default on, what is the opt-out mechanism for users? 
Default Glean SDK opt-out mechanism.
10. Please provide a general description of how you will analyze this data. 
Glean / Amplitude
11. Where do you intend to share the results of your analysis? 
Only on Glean, Amplitude and with mobile teams.
12. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
 No third-party tool will use this data.

**Comment by travis79** [comment]
> ### **Request for data collection review**
> 
>     1. What questions will you answer with this data?
>        More understanding of user behaviour so we can refine suggestions that are shown in the Awesomebar.
> 
>     2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements?    
>        This allows for a better Awesomebar suggestions.
> 
>     3. What alternative methods did you consider to answer these questions? Why were they not sufficient?    
>        No alternative is available to answer this question.
> 
>     4. Can current instrumentation answer these questions?    
>        No current instrumentation can answer this question.
> 
>     5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki.
> 
> 
> Measurement Description 	Data Collection Category 	Tracking Bug #
> Awesomebar bookmark suggestion clicked 	category 2 - interaction data 	#18068
> Awesomebar clipboard suggestion clicked 	category 2 - interaction data 	#18068
> Awesomebar history suggestion clicked 	category 2 - interaction data 	#18068
> Awesomebar search action clicked 	category 2 - interaction data 	#18068
> Awesomebar search suggestion clicked 	category 2 - interaction data 	#18068
> Awesomebar opened tab suggestion clicked 	category 2 - interaction data 	#18068
> 
>     1. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
>        [https://github.com/mozilla-mobile/fenix/blob/master/docs/metrics.md](https://github.com/mozilla-mobile/fenix/blob/master/docs/metrics.md?rgh-link-date=2021-02-19T19%3A27%3A38Z)
> 
>     2. How long will this data be collected? Choose one of the following:    
>        This is scoped to a time-limited experiment/project until date 08-01-2021.
> 
>     3. What populations will you measure?    
>        No filters - all locales, channels, etc.
> 
>     4. If this data collection is default on, what is the opt-out mechanism for users?    
>        Default Glean SDK opt-out mechanism.
> 
>     5. Please provide a general description of how you will analyze this data.    
>        Glean / Amplitude
> 
>     6. Where do you intend to share the results of your analysis?    
>        Only on Glean, Amplitude and with mobile teams.
> 
>     7. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
>            No third-party tool will use this data.

# Data Review

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

Yes, via the metrics.md and other Glean tools.

2) Is there a control mechanism that allows the user to turn the data collection on and off? 

Yes

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

Not applicable, the collection is set to end 2021-08-01.

4) Using the **[category system of data types](https://wiki.mozilla.org/Firefox/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Category 2, User Interaction

5) Is the data collection request for default-on or default-off?

Default on

6) Does the instrumentation include the addition of **any *new* identifiers**?

No

7) Is the data collection covered by the existing Firefox privacy notice?

Yes

8) Does there need to be a check-in in the future to determine whether to renew the data? 

No, unless the owner decides to extend beyond 2021-08-01

9) Does the data collection use a third-party collection tool? 

No

## Result:

**data-review+**

**Review by jonalmeida — APPROVED** [review]
✅

File: app/src/test/java/org/mozilla/fenix/components/metrics/GleanMetricsServiceTest.kt [hunk 1]
```diff
@@ -85,4 +86,31 @@ class GleanMetricsServiceTest {
         assertEquals(1, events[0].extra!!.size)
         assertEquals("APP_ICON", events[0].extra!!["source"])
     }
+
+    @Test
+    fun `awesomebar events is correctly recorded`() {
```
Comment (Amejia481) [inline app/src/test/java/org/mozilla/fenix/components/metrics/GleanMetricsServiceTest.kt:91]: nit: `awesomebar events are correctly recorded`

**Review by Amejia481 — APPROVED** [review]

File: app/src/test/java/org/mozilla/fenix/components/metrics/GleanMetricsServiceTest.kt (hunk 1 above)
Comment (rocketsroger) [reply app/src/test/java/org/mozilla/fenix/components/metrics/GleanMetricsServiceTest.kt:91]: oops.  I'll fix that when I refactor.  thanks for catching it.

**Pull request merged**